src/portable/st/stm32_fsdev/dcd_stm32_fsdev.c

```diff
@@ -666,16 +676,29 @@ void dcd_int_handler(uint8_t rhport) {
   // be triggered repeatedly.
 
   /* Put SOF flag at the beginning of ISR in case to get least amount of jitter if it is used for timing purposes */
-  if(int_status & USB_ISTR_SOF) {
+  if (int_status & USB_ISTR_SOF)
+  {
     USB->ISTR &=~USB_ISTR_SOF;
-    dcd_event_sof(0, USB->FNR & USB_FNR_FN, true);
+    dcd_event_sof(rhport, USB->FNR & USB_FNR_FN, true);
+
+    /* Ensure normal USB operation as a failsafe if a remote wakeup from the host was omitted */
+    USB->CNTR &= ~USB_CNTR_LPMODE;
+    USB->CNTR &= ~USB_CNTR_FSUSP;
+
+    /* Disable SOF if it wasn't enabled to begin with */
+    dcd_sof_enable(rhport, previousSofState);
```

Comment:
There is an issue here, I think.

The call `dcd_sof_enable(rhport, true);` on line 736  sets `previousSofState = true`. But we expect to call `dcd_sof_enable(rhport, false)` here if SOF is used as a failsafe to wake the device?

Reply:
You are right. At 736 either the value should not overwritten or it should be restored.